<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>869</width>
    <height>605</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>（网络化控制）Modbus上位机</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_16">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_9">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <spacer name="verticalSpacer_5">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>串口初始化</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>串口(serial)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_serial"/>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>波特率(baud)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_baud">
              <item>
               <property name="text">
                <string>9600</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>115200</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>数据位(bytesize)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_byteSize">
              <property name="whatsThis">
               <string>字节大小</string>
              </property>
              <item>
               <property name="text">
                <string>8</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>校验位(parity)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_parity">
              <item>
               <property name="text">
                <string>None</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_5">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>停止位(stopbite)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_stopbite">
              <item>
               <property name="text">
                <string>1</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>流控(xonxoff)</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="comboBox_control">
              <item>
               <property name="text">
                <string>0</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="pushButton_linkToSerial">
        <property name="text">
         <string>连接串口</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="Line" name="line_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>显示设置</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QRadioButton" name="radioButton_recvASCII">
              <property name="text">
               <string>BIN</string>
              </property>
              <property name="checked">
               <bool>true</bool>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">Encoding_buttonGroup</string>
              </attribute>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="radioButton_recvHex">
              <property name="text">
               <string>HEX</string>
              </property>
              <attribute name="buttonGroup">
               <string notr="true">Encoding_buttonGroup</string>
              </attribute>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_autoChangeline">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>自动换行</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_showTime">
            <property name="text">
             <string>显示时间</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer_4">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_8">
        <item>
         <widget class="Line" name="line_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>发送设置</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_14">
          <item>
           <widget class="QRadioButton" name="radioButton_sendHex">
            <property name="text">
             <string>HEX</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QCheckBox" name="checkBox_repeat">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>重复发送</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Maximum</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>25</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox_sendTime">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>5000</number>
            </property>
            <property name="singleStep">
             <number>5</number>
            </property>
            <property name="value">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_10">
            <property name="text">
             <string>毫秒每次</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_10">
          <item>
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>读取个数</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_readNum">
            <item>
             <property name="text">
              <string>1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>slave</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_slave">
            <item>
             <property name="text">
              <string>1</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>2</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_12">
          <item>
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>功能码</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="comboBox_funcCode">
            <item>
             <property name="text">
              <string>READ_HOLDING_REGISTERS</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>READ_COILS</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>READ_DISCRETE_INPUTS</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>READ_INPUT_REGISTERS</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>WRITE_SINGLE_COIL</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>WRITE_SINGLE_REGISTER</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>WRITE_MULTIPLE_COILS</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>WRITE_MULTIPLE_REGISTERS </string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="checkBox_show2data">
            <property name="text">
             <string>显示两个数据</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_6">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <widget class="QTabWidget" name="tabWidget_view">
        <property name="contextMenuPolicy">
         <enum>Qt::DefaultContextMenu</enum>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <property name="usesScrollButtons">
         <bool>true</bool>
        </property>
        <property name="documentMode">
         <bool>false</bool>
        </property>
        <widget class="QWidget" name="tabWidgetPage_text">
         <attribute name="title">
          <string>数据接收</string>
         </attribute>
        </widget>
        <widget class="QWidget" name="tabWidgetPage_plot">
         <attribute name="title">
          <string>绘制折线图</string>
         </attribute>
        </widget>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer_2">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Fixed</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>10</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_15">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_13">
            <item>
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>地址一</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit_addr">
              <property name="text">
               <string>0x2010</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>地址二</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit_addr2">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="text">
               <string>0x2010</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_9">
            <item>
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>对应的二进制:</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit_addrTransTobin">
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="lineEdit_lineEdit_addrTransTobin2">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>18</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="pushButton_send">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>发送</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>文件保存路径：</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_saveFilePath">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_selectFileSavePath">
          <property name="text">
           <string>选择文件夹路径</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_save">
          <property name="text">
           <string>保存数据</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>869</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>comboBox_serial</tabstop>
  <tabstop>comboBox_baud</tabstop>
  <tabstop>comboBox_byteSize</tabstop>
  <tabstop>comboBox_parity</tabstop>
  <tabstop>comboBox_stopbite</tabstop>
  <tabstop>comboBox_control</tabstop>
  <tabstop>radioButton_recvASCII</tabstop>
  <tabstop>radioButton_recvHex</tabstop>
  <tabstop>checkBox_autoChangeline</tabstop>
  <tabstop>checkBox_showTime</tabstop>
  <tabstop>radioButton_sendHex</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>checkBox_repeat</sender>
   <signal>clicked(bool)</signal>
   <receiver>spinBox_sendTime</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>492</y>
    </hint>
    <hint type="destinationlabel">
     <x>227</x>
     <y>494</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_linkToSerial</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_serial</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>197</x>
     <y>264</y>
    </hint>
    <hint type="destinationlabel">
     <x>290</x>
     <y>80</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_linkToSerial</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_baud</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>182</x>
     <y>258</y>
    </hint>
    <hint type="destinationlabel">
     <x>222</x>
     <y>112</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_linkToSerial</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_byteSize</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>96</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>228</x>
     <y>140</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_linkToSerial</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_parity</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>143</x>
     <y>268</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_linkToSerial</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_stopbite</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>262</y>
    </hint>
    <hint type="destinationlabel">
     <x>233</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_linkToSerial</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBox_control</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>269</x>
     <y>256</y>
    </hint>
    <hint type="destinationlabel">
     <x>265</x>
     <y>227</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_linkToSerial</sender>
   <signal>toggled(bool)</signal>
   <receiver>pushButton_send</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>326</x>
     <y>257</y>
    </hint>
    <hint type="destinationlabel">
     <x>854</x>
     <y>531</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_linkToSerial</sender>
   <signal>clicked(bool)</signal>
   <receiver>checkBox_repeat</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>35</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>50</x>
     <y>484</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="Encoding_buttonGroup"/>
 </buttongroups>
</ui>
